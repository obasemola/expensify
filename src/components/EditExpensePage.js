import React from 'react';

const EditExpensePage = () => (
  <div>
    This where I edit stuff
  </div>
);

export default EditExpensePage