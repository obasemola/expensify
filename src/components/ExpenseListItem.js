import React from 'react';


const ExpenseListItem = ({ description, amount, createdAt }) => (
  <div>
    <p>{description}</p>
    <p>{amount}</p>
    <p>{createdAt}</p>
  </div>
)

export default ExpenseListItem;