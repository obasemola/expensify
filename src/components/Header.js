import React from 'react';
import { NavLink } from 'react-router-dom';


const Header = () => (
  <header>
    <h1>Expensify</h1>
    <NavLink to='/'>Go to the home page</NavLink>
    <NavLink to='/create'>Go to the create page</NavLink>
    <NavLink to='/help'>Go to the help page</NavLink>
  </header>
)

export default Header;