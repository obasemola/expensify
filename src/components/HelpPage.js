import React from 'react';

const HelpPage = () => (
  <div>
    You need some help?
  </div>
);

export default HelpPage