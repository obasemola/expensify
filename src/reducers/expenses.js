const expenseReducerDefaultState = [];

export default ( state = expenseReducerDefaultState, action) => {
  switch (action.type) {
    case 'ADD_EXPENSE':
      return [
        ...state, action.expense
      ]

    case 'REMOVE_EXPENSE':
      return state.filter(el => el.id !== action.id);

    case 'EDIT_EXPENSE':
      return state.map((expense) => {
        if(expense.id === action.id){
          return {
            ...expense,
            ...action.updates
          } 
        } else {
          return
        }
      })

    default:
      return state
  }
};