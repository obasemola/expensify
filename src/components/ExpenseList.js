import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import selectedExpenses from '../selectors/filterExpenses'

export const ExpenseList = (props) => (
  <div>
    {
      props.expenses.length === 0 ?
      <p>No expenses</p> :
      props.expenses.map((expense) => {
        return <ExpenseListItem expense={expense} key={expense.id} {...expense} id={expense.id}/>
      })
    }
  </div>
);


const mapStateToProps = (state) => {
  return {
    expenses: selectedExpenses(state.expenses, state.filters)
  }
}

export default connect(mapStateToProps)(ExpenseList)