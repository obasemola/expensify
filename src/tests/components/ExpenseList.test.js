import React from 'react';
import { shallow } from 'enzyme';
import { ExpenseList } from '../../components/ExpenseList';
import expenses from '../fixtures/expenses';

describe('should render expense list items', () => {
  test('should render expense list with expenses', () => {
    const wrapper = shallow(<ExpenseList expenses={expenses}/>);

    expect(wrapper).toMatchSnapshot();
  })

  test('should render expense list with no expenses', () => {
    const wrapper = shallow(<ExpenseList expenses={[]}/>);

    expect(wrapper).toMatchSnapshot();
  })
})