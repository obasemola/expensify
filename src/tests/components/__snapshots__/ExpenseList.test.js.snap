// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render expense list items should render expense list with expenses 1`] = `
<div>
  <ExpenseListItem
    amount={195}
    createdAt={0}
    description="Gum"
    expense={
      Object {
        "amount": 195,
        "createdAt": 0,
        "description": "Gum",
        "id": "1",
        "note": "",
      }
    }
    id="1"
    key="1"
    note=""
  />
  <ExpenseListItem
    amount={19500}
    createdAt={-345600000}
    description="Rent"
    expense={
      Object {
        "amount": 19500,
        "createdAt": -345600000,
        "description": "Rent",
        "id": "2",
        "note": "",
      }
    }
    id="2"
    key="2"
    note=""
  />
  <ExpenseListItem
    amount={4500}
    createdAt={345600000}
    description="Credit card"
    expense={
      Object {
        "amount": 4500,
        "createdAt": 345600000,
        "description": "Credit card",
        "id": "1",
        "note": "",
      }
    }
    id="1"
    key="1"
    note=""
  />
</div>
`;

exports[`should render expense list items should render expense list with no expenses 1`] = `
<div>
  <p>
    No expenses
  </p>
</div>
`;
